shared/src/ledger/signing.rs

```diff
@@ -277,10 +277,11 @@ pub async fn aux_signing_data<
     };
 
     if fee_payer == masp_tx_key().to_public() {
-        panic!(
+        return Err(Error::Tx(TxError::Other(
```

Comment:
For this one it may be god to just return a Error::Other. Though it doesn't matter too much.